src/main/java/com/example/demo/OrderComponent/Domain/Services/OrderService.java

```diff
@@ -0,0 +1,214 @@
+    }
+
+    @Transactional
+    public OrderDTO modifyOrder(UUID orderId, ModifyOrderDTO modifyOrderRequest) {
+        Order order = orderRepository.findById(orderId)
+                .orElseThrow(() -> new EntityNotFoundException("Order not found"));
+
+        if (order.getStatus() == OrderStatus.CLOSED) {
+            throw new IllegalStateException("Order is already closed and cannot be modified.");
+        }
+
+        if (modifyOrderRequest.getReservationId() != null) {
+            order.setReservationId(modifyOrderRequest.getReservationId());
+        }
+
+        if (modifyOrderRequest.getStatus() != null) {
+            order.setStatus(modifyOrderRequest.getStatus());
+        }
+
+        List<OrderItem> existingItems = orderItemRepository.findByOrderId(order.getId());
+        Map<UUID, OrderItem> existingItemsMap = existingItems.stream()
+                .collect(Collectors.toMap(OrderItem::getId, item -> item));
+
+        BigDecimal originalPrice = BigDecimal.ZERO;
+        Currency currency = Currency.EUR;
+        List<OrderItem> updatedItems = new ArrayList<>();
+
+        if (modifyOrderRequest.getItems() != null) {
+            for (OrderItemDTO itemRequest : modifyOrderRequest.getItems()) {
+                OrderItem orderItem = existingItemsMap.get(itemRequest.getProductId());
+                boolean isNewItem = orderItem == null;
+
+                if (isNewItem) {
+                    orderItem = new OrderItem();
+                    orderItem.setOrderId(order.getId());
+                }
+
+                orderItem.setProductID(itemRequest.getProductId());
+                orderItem.setTitle(itemRequest.getTitle());
+                orderItem.setQuantity(itemRequest.getQuantity());
+                orderItem.setUnitPrice(itemRequest.getUnitPrice().getBase());
+                orderItem.setCurrency(itemRequest.getCurrency());
+
+                originalPrice = originalPrice.add(orderItem.getUnitPrice().multiply(BigDecimal.valueOf(orderItem.getQuantity())));
+
+                OrderItem savedOrderItem = orderItemRepository.save(orderItem);
+                updatedItems.add(savedOrderItem);
+
+                if (itemRequest.getModifiers() != null) {
+                    for (OrderItemModifierDTO modifierDTO : itemRequest.getModifiers()) {
+                        OrderItemModifier orderItemModifier = new OrderItemModifier();
+                        orderItemModifier.setOrderItemId(savedOrderItem.getId());
+                        orderItemModifier.setProductModifierId(modifierDTO.getId());
+                        orderItemModifier.setTitle(modifierDTO.getTitle());
+                        orderItemModifier.setPrice(modifierDTO.getPrice());
+                        orderItemModifier.setCurrency(modifierDTO.getCurrency());
+                        orderItemModifierRepository.save(orderItemModifier);
+                    }
+                }
+            }
+        }
+
```

Comment:
This method definitely does not fit into one window. Break it up, use mappers, ect. It's too long!